dcim_related-connections.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 The go-netbox Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netbox
+
+import (
+	"errors"
+	"net/http"
+	"net/url"
+)
+
+// RelatedConnection represents components that have a releated peer-device and
+// peer-interface
+type RelatedConnection struct {
+	Device       *Device                    `json:"device"`
+	ConsolePorts []*RCConsolePortIdentifier `json:"console-ports"`
+	Interfaces   []*RCInterfaceIdentifier   `json:"interfaces"`
+	PowerPorts   []*RCPowerPortIdentifier   `json:"power-ports"`
+}
+
+// RCConsolePortIdentifier represents a reduced version of a console port
+type RCConsolePortIdentifier struct {
+	ConsoleServer string `json:"console-server"`
+	Name          string `json:"name"`
+	Port          string `json:"port"`
+}
+
+// RCInterfaceIdentifier represents a reduced version of a device interface
+type RCInterfaceIdentifier struct {
+	Device    string `json:"device"`
+	Interface string `json:"interface"`
+	Name      string `json:"name"`
+}
+
+// RCPowerPortIdentifier represents a reduced version of a single power port
+type RCPowerPortIdentifier struct {
+	PDU    string `json:"pdu"`
+	Name   string `json:"name"`
+	Outlet string `json:"outlet"`
+}
+
+// GetRelatedConnections retreives a RelatedConnections objects from Netbox
+func (s *DCIMService) GetRelatedConnections(
+	options *RelatedConnectionsOptions,
```

Comment:
If the only two parameters for this API request are `peer-device` and `peer-interface`, and they are mandatory, I say we change the signature to:

```go
func (s *DCIMService) GetRelatedConnections(peerDevice, peerInterface string) (*RelatedConnection, error)
```